### ISSUE TITLE
Region 82 Avg IOU: -nan, Class:-nan, Obj:-nan....., when training on my dataset

### ISSUE THREAD

**Issue by jt387** [pjreddie/darknet#1117]
Hi, all！ I use yolov3 to detect the traffic light, the resolution of the picture is 1292*964, and the height range of the traffic light is about between 30 pixels and 60 pixels,  the weight range of the traffic light is about between 10 pixels and 30 pixels.  
And I modify the yolov3-voc.cfg as below:
line 6:   batch=8
line 7:   subdivisions=4
line610, 694, 778 anchors=4.79,12.52, 5.50,16.05, 6.63,21.97, 7.52,16.18, 8.67,24.09, 9.45,8.02, 12.70,11.94, 13.69,9.11, 15.00,12.69, 16.98,13.18
 I use this [https://github.com/B1SA/smbmkt/blob/master/detector/yolo/training/generate_anchors_yolo_v3.py](url) to recalculate the anchors of my dataset
line 616, 700, 784 random=0
And I use this command to train the model: ./darknet detector train cfg/voc.data cfg/yolov3_trafficLight.cfg darknet53.conv.74,  the display is as below:
![image](https://user-images.githubusercontent.com/6219009/45418676-a77c1f00-b6b7-11e8-9ac8-92f033ee8a7d.png)

Why does this problem appear? The traffic light is too small? How to solve this problem? Thanks!

**Comment by jt387** [comment]
I copied  yolov3-voc.cfg to yolov3_trafficLight.cfg, so the training command uses the right configure file

**Comment by hajlyx** [comment]
Your batch size is too low, set batch size to 16 or above.

**Comment by zhaonann** [comment]
Hi, I meet the same problem, and I tried to change the value of batch size. Finally, everything is not happened, the problem is still before. Why and how can I solve it?

**Comment by barzan-hayati** [comment]
I decrease learning rate from 0.001 to 0.0001 and it worked for me.

**Comment by zhaonann** [comment]
Thank you for your reply. It works.发自我的iPhone------------------ Original ------------------From: barzan-hayati  <notifications@github.com>Date: Sat,Aug 3,2019 8:50 PMTo: pjreddie/darknet <darknet@noreply.github.com>Cc: zhaonann <sailonzn@163.com>, Comment <comment@noreply.github.com>Subject: Re: [pjreddie/darknet] Region 82 Avg IOU: -nan, Class:-nan,Obj:-nan....., when training on my dataset (#1117)I decrease learning rate from 0.001 to 0.0001 and it worked for me.

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or mute the thread.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/pjreddie/darknet/issues/1117?email_source=notifications\u0026email_token=AJXE4ITBN3S73Q3BVVOS4OLQCV5JZA5CNFSM4FUUJXV2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD3PN4OA#issuecomment-517922360",
"url": "https://github.com/pjreddie/darknet/issues/1117?email_source=notifications\u0026email_token=AJXE4ITBN3S73Q3BVVOS4OLQCV5JZA5CNFSM4FUUJXV2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD3PN4OA#issuecomment-517922360",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
